apps/api-extractor/src/analyzer/PackageMetadataManager.ts

```diff
@@ -70,37 +200,30 @@ export class PackageMetadataManager {
     this._messageRouter = messageRouter;
   }
 
-  // This feature is still being standardized: https://github.com/microsoft/tsdoc/issues/7
-  // In the future we will use the @microsoft/tsdoc library to read this file.
+  /**
+   * This feature is still being standardized: https://github.com/microsoft/tsdoc/issues/7
+   * In the future we will use the @microsoft/tsdoc library to read this file.
+   */
   private static _resolveTsdocMetadataPathFromPackageJson(
```

Comment:
Currently, `_resolveTsdocMetadataPathFromPackageJson()` is used in two different ways:

- **reading:** When API Extractor analyzes an external NPM package, this function is used to probe for `tsdoc-metadata.json`.  If found, that file indicates that the external NPM package was built with API Extractor and its .d.ts files `/** */` comments can safely be interpreted as TSDoc/AEDoc for the purposes of features like .d.ts trimming, `{@inheritDoc}`, etc.

- **writing:** When API Extractor is used to analyze an local project, it generates a `tsdoc-metadata.json` file to be published with the NPM package.  If [tsdocMetadataFilePath](https://api-extractor.com/pages/configs/api-extractor_json/#tsdocmetadatatsdocmetadatafilepath) is not specified, it defaults to `<lookup>` which guesses where to write the file.  This guess reuses the exact same rules as `_resolveTsdocMetadataPathFromPackageJson()`.

However, if the lookup process is getting more complex and now includes considerations like `typesVersions`, maybe it is not the best design to use identical rules for reading/writing.  Consider if instead we did something like:

- **reading:** Look in all the reasonable places, from most specific (by the entry point .d.ts file) to least specific (in the package.json root)
- **writing:** Use a more simple policy that chooses a folder that is most likely to be guessed correctly by the "reading" logic.  For really complex cases (e.g. multiple entry points and multiple compiler versions) maybe the user is better off specifying `tsdocMetadataFilePath` manually?

I don't insist to address that in this PR, but it's worth thinking about in this moment where we've remembered all the details about this feature.